classes/CMSCategory.php

```diff
@@ -246,48 +246,50 @@ protected function recursiveDelete(&$to_delete, $id_cms_category)
         }
     }
 
+    /**
+     * Directly call the parent of delete, in order to avoid recursion
+     *
+     * @return bool Deletion result
+     */
+    public function deleteLite()
+    {
+        return parent::delete();
+    }
+
     public function delete()
     {
-        if ($this->id == 1) {
+        if ((int)$this->id === 1) {
             return false;
         }
 
         $this->clearCache();
 
-        // Get children categories
-        $to_delete = array((int)$this->id);
-        $this->recursiveDelete($to_delete, (int)$this->id);
-        $to_delete = array_unique($to_delete);
-
-        // Delete CMS Category and its child from database
-        $list = count($to_delete) > 1 ? implode(',', $to_delete) : (int)$this->id;
-        $id_shop_list = Shop::getContextListShopID();
-        if (count($this->id_shop_list)) {
-            $id_shop_list = $this->id_shop_list;
+        $objects = $this->getAllChildren();
```

Comment:
I don't like `objects` as the variable name, we are manipulating CMS categories here, right?

So I'd like to the see `$cmsCategories` instead.